src/main/java/ru/test/restservice/api/FileController.java

```diff
@@ -0,0 +1,103 @@
+package ru.test.restservice.api;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.test.restservice.domain.FileList;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+@RestController
+@RequestMapping("/api")
+public class FileController {
+
+    // Список разрешённых файловых расширений
+    final ArrayList<String> maskTextFile = new ArrayList<>(Arrays.asList(".bib", ".tex"));
+    final ArrayList<String> maskPic = new ArrayList<>(Arrays.asList(".png", ".jpg", ".svg"));
+    final ArrayList<String> mask = joinLists(maskPic, maskTextFile);
+
+    // Функция объединения списков (надо бы вынести в класс utils)
+    public <T> ArrayList<T> joinLists(
+            final List<T> listA,
+            final List<T> listB) {
+        boolean aEmpty = (listA == null) || listA.isEmpty();
+        boolean bEmpty = (listB == null) || listB.isEmpty();
+        if (aEmpty & bEmpty) {
+            return new ArrayList<>();
+        } else if (aEmpty) {
+            return new ArrayList<>(listB);
+        } else if (bEmpty) {
+            return new ArrayList<>(listA);
+        } else {
+            ArrayList<T> result = new ArrayList<>(
+                    listA.size() + listB.size());
+            result.addAll(listA);
+            result.addAll(listB);
+            return result;
+        }
+    }
+
+    // Ловлю передаваемые файлы, создаю их в файловой системе и возвращаю соответствующий http код
+    @PostMapping(value = "/upload", headers = "content-type=multipart/*")
+    public ResponseEntity<?> handleFileUpload(@RequestParam("file") MultipartFile file) {
+        try {
+            // Проверяю и на фронте, и на бэке разрешён ли файл (на фронте пользователь может удалить метод)
+            boolean isAllowedExtension = false;
+            for (String s : mask) {
+                isAllowedExtension = isAllowedExtension || Objects.requireNonNull(file.getOriginalFilename()).toLowerCase().endsWith(s);
+            }
```

Comment:
Тут стримы приходят на помощь
```suggestion
            String name = file.getOriginalFilename().toLowerCase();
            boolean isAllowedExtension = mask.stream().anyMatch(m -> name.endsWith(m));
```

Reply:
Почитаю про это, спасибо, но я решил регулярками